src/Data/List/Properties.agda

```diff
@@ -842,6 +842,27 @@ sum-++ (x ∷ xs) ys = begin
 ∈⇒∣product {n} {n ∷ ns} (here  refl) = divides (product ns) (*-comm n (product ns))
 ∈⇒∣product {n} {m ∷ ns} (there n∈ns) = ∣n⇒∣m*n m (∈⇒∣product n∈ns)
 
+nonZero-product : ∀ {ns} → All NonZero ns → NonZero (product ns)
+nonZero-product [] = _
+nonZero-product {n ∷ ns} (nzn ∷ nzns) = m*n≢0 n (product ns)
+  where instance
+    _ = nzn
+    _ = nonZero-product nzns
+
+∈⇒≤product : ∀ {n ns} → n ∈ ns → All NonZero ns → n ≤ product ns
+∈⇒≤product {n} {m ∷ ms} (here n≡m) (_ ∷ nzms) rewrite n≡m =
+  m≤m*n m (product ms)
+  where instance _ = nonZero-product nzms
+∈⇒≤product {n} {m ∷ ms} (there n∈ns) (nz ∷ nzns) = begin-ordered
+  n               ≤⟨ ∈⇒≤product n∈ns nzns ⟩
+  product ms      ≤⟨ m≤n*m (product ms) m ⟩
+  m * product ms  ∎-ordered
+  where
+  instance _ = nz
+  open ≤-Reasoning using (step-≤)
+    renaming ( begin_ to begin-ordered_ ; _∎ to _∎-ordered)
```

Comment:
Oh dear, yes, that `renaming` is a bit heavy/unpleasant!
Would
```agda
renaming ( begin_ to begin-≤_ ; _∎ to _≤-∎)
```
be any better?

Reply:
I agree that this is too heavyweight. Probably back to the `trans` version is going to be cleanest.

Reply:
Or: as indicated above: localise the reasoning with `ℕ._≤_` and `ℕ._*_` in `Data.Nat.Properties`, and then appeal to it when proving properties of `Data.List.Base.product` in `Data.List.Properties`... noting that the lemmas I propose above are 'hidden' appeals to transitivity on top of the existing `m≤m*n` etc...